File: s11_Retrofit/KriptoPare/app/src/main/java/com/omeryildizce/kriptopara/adapter/CryptoAdapter.java
package com.omeryildizce.kriptopara.adapter;

import android.graphics.Color;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;

import com.omeryildizce.kriptopara.R;
import com.omeryildizce.kriptopara.model.CryptoModel;

import java.util.ArrayList;

public class CryptoAdapter extends RecyclerView.Adapter<CryptoAdapter.CryptoHolder> {
    private ArrayList<CryptoModel> cryptoModelList;
    private String[] colors = {"#F44336", "#E91E63", "#9C27B0", "#673AB7",
            "#3F51B5", "#2196F3", "#03A9F4", "#00BCD4",
            "#009688", "#4CAF50", "#8BC34A", "#CDDC39",
            "#FFEB3B", "#FFC107", "#FF9800", "#FF5722"
    };

    public CryptoAdapter(ArrayList<CryptoModel> cryptoModelList) {
        this.cryptoModelList = cryptoModelList;

    }

    @NonNull
    @Override
    public CryptoHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        View view = layoutInflater.inflate(R.layout.row_layout, parent, false);

        return new CryptoHolder(view);

    }

    @Override
    public void onBindViewHolder(@NonNull CryptoHolder holder, int position) {
        holder.bind(cryptoModelList.get(position), colors, position);
    }

    @Override
    public int getItemCount() {
        return cryptoModelList.size();
    }

    public class CryptoHolder extends RecyclerView.ViewHolder {
        TextView cryptoName;
        TextView cryptoPrice;

        public CryptoHolder(@NonNull View itemView) {
            super(itemView);


        }

        public void bind(CryptoModel cryptoModel, String[] colors, int position) {
            CardView cryptoCard = itemView.findViewById(R.id.cryptoCard);
            cryptoCard.setCardBackgroundColor(Color.parseColor(colors[position % 16]));
            cryptoName = itemView.findViewById(R.id.cryptoName);
            cryptoPrice = itemView.findViewById(R.id.cryptoPrice);
            cryptoName.setText(cryptoModel.currency);
            cryptoPrice.setText(cryptoModel.price);

        }

    }
}

File: s24_Retrofit/KriptoPara/app/src/main/java/com/omeryildizce/kriptopara/adapter/CryptoAdapter.kt
package com.omeryildizce.kriptopara.adapter

import android.graphics.Color
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import androidx.viewbinding.ViewBinding
import com.omeryildizce.kriptopara.databinding.RowLayoutBinding
import com.omeryildizce.kriptopara.model.CryptoModel

class CryptoAdapter(val cryptoModelList: ArrayList<CryptoModel>,   val listener:Listener) :RecyclerView.Adapter<CryptoAdapter.CryptoHolder>() {
    private val colors = arrayOf("#FF0000", "#00FF00", "#0000FF", "#FFFF00", "#FF00FF", "#00FFFF", "#800000", "#008000", "#000080", "#808000", "#800080", "#008080", "#C0C0C0", "#808080", "#FFA07A", "#20B2AA", "#FFC0CB", "#7B68EE", "#F0E68C", "#DDA0DD")
    interface Listener{
        fun onItemCick(cryptoModel: CryptoModel)
    }
    class CryptoHolder(var binding: RowLayoutBinding) : RecyclerView.ViewHolder(binding.root)  {
        fun bind(cryptoModel: CryptoModel, colors:Array<String> , position: Int, listener:Listener){
             binding.textViewName.text = cryptoModel.currency
             binding.textViewPrice.text = cryptoModel.price
             binding.linearLayout.setBackgroundColor(Color.parseColor(colors[position%20]))
            itemView.setOnClickListener {
                listener.onItemCick(cryptoModel)
            }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CryptoHolder {
        val binding = RowLayoutBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return CryptoHolder(binding)
    }

    override fun getItemCount(): Int {
        return cryptoModelList.size
    }

    override fun onBindViewHolder(holder: CryptoHolder, position: Int) {
        holder.bind(cryptoModelList.get(position), colors, position,listener )

    }
}